web/src/rendered_markdown.ts

```diff
@@ -245,10 +262,15 @@ export const update_elements = ($content: JQuery): void => {
                 // If the stream has been deleted,
                 // sub_store.maybe_get_stream_name might return
                 // undefined.  Otherwise, display the current stream name.
-                const text = $(this).text();
-                $(this).text("#" + stream_name + text.slice(text.indexOf(" > ")));
+                const stream_topic_html = $(this).html();
+                const new_stream_topic_html =
+                    `#${_.escape(stream_name)}` +
+                    stream_topic_html.slice(stream_topic_html.indexOf(" &gt; "));
+                $(this).html(new_stream_topic_html);
             }
         }
+
+        maybe_update_empty_string_topic_element($(this));
```

Comment:
We'll end up doing some restructuring here based on the [CZO Discussion](https://chat.zulip.org/#narrow/channel/378-api-design/topic/notation.20for.20near.20links/near/2066034) I posted.

Reply:
Yeah, we should be able to rework what you have to start with parsing the URL in all cases. Note also that we want to handle the stream being renamed here, so I think rather than adding a new function, it'll just be a rewrite of this block to use the modern practices of parsing the URL, then looking up the stream name by ID and possibly modifying the topic, and then writing the HTML out.

Reply:
Done